core/modules/node/node.types.inc

```diff
@@ -287,12 +287,27 @@ function node_type_form($form, &$form_state, $type = NULL) {
   );
 
   $form['actions'] = array('#type' => 'actions');
+  // If this is a new content type and the Field UI module is enabled, add a
+  // 'Save and add fields' button.
+  if ($type->module === 'node' && empty($type->type) && module_exists('field_ui')) {
+    $form['actions']['add_fields'] = array(
```

Comment:
In general the thing that matters the most, changes the least amount of code and accomplishes the use case is @quicksketch 's recommendation:  altering this function call `foreach (element_children($element, TRUE) as $key) {` of `element_children` to include the `TRUE` paramater.

I think we should go with that solt'n.  then the only thing that needs to change is adding the `TRUE` paramater and changing the weight of the form element in  `field_ui.module`:

```php
/**
 * Implements hook_form_FORM_ID_alter().
 *
 * Adds a button 'Save and add fields' to the 'Create content type' form.
 *
 * @see node_type_form()
 * @see field_ui_form_node_type_form_submit()
 */
function field_ui_form_node_type_form_alter(&$form, $form_state) {
  // We want to display the button only on add page.
  if (empty($form['#node_type']->type)) {
    $form['actions']['save_continue'] = array(
      '#type' => 'submit',
      '#value' => t('Save and add fields'),
      '#weight' => 45,
    );
    $form['#submit'][] = 'field_ui_form_node_type_form_submit';
  }
}
````